### ISSUE TITLE
[Identify subjects] Doesn't work with BehaviorSubject, EtcSubject 

### ISSUE THREAD

**Issue by splincode** [cartant/eslint-plugin-rxjs#88]
```json
{
  "rxjs/suffix-subjects": [
    "error",
    {
      "parameters": true,
      "properties": true,
      "suffix": "$$",
      "variables": true,
    }
  ]
}
```

```ts
const answers1 = new BehaviorSubject<number>(0); // no eslint error

class MySuperSubject extends Subject {}
class MySuperBehaviorSubject extends BehaviorSubject {}

const answers2 = new MySuperSubject<number>(); // no eslint error
const answers3 = new MySuperBehaviorSubject<number>(0); // no eslint error
```

**Comment by cartant** [comment]
Works in the tests I just added:

https://github.com/cartant/eslint-plugin-rxjs/blob/695b783fc015ee19c9d9e3a20c89b0180b888911/tests/rules/suffix-subjects.ts#L585-L603

Also, this is not valid, as `Subject` has type parameters:

```ts
class MySuperSubject extends Subject {}
```

**Comment by splincode** [comment]
what about test suffix is "$$", when I have `const answer$ = new BehaviorSubject();`?

**Comment by cartant** [comment]
The `suffix` option is escaped in 4.0.3, so the `$$` suffix thing should be fixed in that version.

**Comment by cotneit** [comment]
> what about test suffix is "$$", when I have `const answer$ = new BehaviorSubject();`?

Same config, including "$$" suffix
In 5.0.1 (also tried 4.0.3 and 4.0.4 with the same results) this seems to still be an issue for class properties

```ts
const destroy$: Subject<void> = new Subject<void>(); // eslint error

const obj = {
    destroy$: new Subject<void>(), // eslint error
}

function fn(destroy$: Subject<void>) {} // eslint error

export class OrderListComponent {
    private destroy: Subject<void> = new Subject<void>(); // eslint error
    private destroy$: Subject<void> = new Subject<void>(); // no eslint error   <---
}
```

**Comment by cartant** [comment]
@cotneit Should be fixed in 5.0.2. In future, please open a new issue instead of commenting on a closed issue. You can reference the closed issue from the new issue. In general, that's something you should do for all repos - not just this one. It's easy to overlook such comments.

**Comment by cotneit** [comment]
@cartant Thank you! Appreciate the advice, will do so in the future!
